engine/src/player-platform.cpp

```diff
@@ -881,6 +882,7 @@ MCPlayer::MCPlayer(const MCPlayer &sref) : MCControl(sref)
 {
 	nextplayer = NULL;
 	filename = strclone(sref.filename);
+    resolved_filename = strclone(sref.resolved_filename);
```

Comment:
Don't clone the resolved_filename - it should be reinitialized at the appropriate time... A cloned object won't necessarily be on the same stack and so the resolved_filename might be different.